pkg/cmd/grafana-cli/commands/upgrade_all_command.go

```diff
@@ -49,19 +49,32 @@ func (cmd Command) upgradeAllCommand(c utils.CommandLine) error {
 		}
 	}
 
+	if len(pluginsToUpgrade) == 0 {
+		return nil
+	}
+
+	var upgraded = false
 	for _, p := range pluginsToUpgrade {
 		logger.Infof("Updating %v \n", p.ID)
 
 		err := services.RemoveInstalledPlugin(pluginsDir, p.ID)
 		if err != nil {
+			if upgraded {
+				warnAboutRestart()
+			}
 			return err
 		}
 
 		err = installPlugin(context.Background(), p.ID, "", c)
 		if err != nil {
+			if upgraded {
+				warnAboutRestart()
```

Comment:
And if you do need to restart if you upgrade plugins, you'll need to do it even if you only manage to upgrade some of them...